R/tools.R

```diff
@@ -139,11 +164,24 @@ rhino_style <- function() {
 #'   format_r("app/view")
 #' }
 #' @export
-format_r <- function(paths) {
+format_r <- function(paths, exclude_files = c("__init__\\.R")) {
+  style_box_use <- box.linters::style_box_use_dir
+  if (!box.linters::is_treesitter_installed()) {
+    style_box_use <- function(path) { }
+    cli::cli_warn(
+      c(
+        "x" = "The packages {{treesitter}} and {{treesitter.r}} are required by some features of `format_r()`.", #nolint
+        "i" = "These package require R version >= 4.3.0 to install."
+      )
+    )
+  }
+
   for (path in paths) {
     if (fs::is_dir(path)) {
-      styler::style_dir(path, style = rhino_style)
+      style_box_use(path, exclude_files = exclude_files)
+      styler::style_dir(path, style = rhino_style, exclude_files = exclude_files)
     } else {
+      style_box_use(path, exclude_files = exclude_files)
```

Comment:
Support for `exclude_files` added to both `box.linters` styling and `styler` styling.